cloudtrail-eventbridge-lambda-dynamodb-sam/src/function.py

```diff
@@ -0,0 +1,49 @@
+import json
+import boto3
+import os
+
+ec2 = boto3.client('ec2')
+dynamodb = boto3.resource("dynamodb")
+table = dynamodb.Table(os.environ["DynamoDBTable"])
+sns = boto3.client('sns')
+
+def lambda_handler(event, context):
+    print(json.dumps(event))
+    print(event['detail']['eventName'], " requested for ", event['detail']['requestParameters']['AdvertiseByoipCidrRequest']['Cidr'])
+    cidr = event['detail']['requestParameters']['AdvertiseByoipCidrRequest']['Cidr']
+    allowed = False
+    if (event['detail']['eventName'] == 'AdvertiseByoipCidr'):
+        allowed_cidrs = table.get_item(
+            Key={
+                'id': 'Allowed_CIDRs'
+            }
+        )
+        print (allowed_cidrs)
+        try:
+            for allowed_cidr in allowed_cidrs['Item']['CIDRs']:
+                if (allowed_cidr == cidr):
```

Comment:
You are just checking for equality here but what happens if the advertised range is just a subset of the ones in the table?